File: test-suite/src/test/java/io/micronaut/docs/validation/iterable/BookInfo.java
package io.micronaut.docs.validation.iterable;

import java.util.List;
import java.util.Map;

// tag::object[]

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;

public class BookInfo {
    private List<@NotBlank String> authors; // <1>

    private Map<@NotBlank String, @Min(1) Integer> sectionStartPages; // <2>
}

// end:object[]

File: test-suite-python/src/test/python/micronaut/docs/validation/iterable/BookInfo.py
from dataclasses import dataclass, field
from typing import Annotated

# tag::object[]

from jakarta.validation.constraints import Min, NotBlank


@dataclass
class BookInfo:
    authors: list[Annotated[str, NotBlank]] = field(default_factory=list)  # <1>

    section_start_pages: dict[Annotated[str, NotBlank], Annotated[int, Min(1)]] = field(default_factory=dict)  # <2>

# end::object[]
